Indexing/TermIndex.cpp

```diff
@@ -177,11 +143,13 @@ void DemodulationLHSIndex::handleClause(Clause* c, bool adding)
   Literal* lit=(*c)[0];
   TermIterator lhsi=EqHelper::getDemodulationLHSIterator(lit, true, _ord, _opt);
   while (lhsi.hasNext()) {
+    auto lhs = lhsi.next();
+    TypedTermList tt(lhs, SortHelper::getTermSort(lhs, lit));
```

Comment:
This is very expensive. We should change getDemodulationLHSIterator to retur a TypedTermList straight away. 
The same pattern is used in many places.